src/astro/sql/operators/sql_decorator.py

```diff
@@ -22,6 +26,127 @@
 from astro.utils.table_handler import TableHandler
 
 
+class SqlExecutor(BaseOperator):
+    def __init__(
+        self,
+        sql: Optional[str] = None,
+        conn_id: Optional[str] = None,
+        database: Optional[str] = None,
+        schema: Optional[str] = None,
+        role: Optional[str] = None,
+        warehouse: Optional[str] = None,
+        parameters: Optional[dict] = None,
+        handler: Optional[Callable] = None,
+        **kwargs,
+    ):
+        self.conn_id = conn_id
+        self.database = database
+        self.schema = schema
+        self.role = role
+        self.warehouse = warehouse
+        self.parameters = parameters
+        self.sql = sql
+        self.handler = handler
+        self.kwargs = kwargs or {}
+
+        if self.parameters is None:
+            self.parameters = {}
+
+        self.op_kwargs: Dict = self.kwargs.get("op_kwargs") or {}
+
+        super().__init__(
+            **kwargs,
+        )
+
+    @property
+    def hook(self):
+        return get_hook(
+            conn_id=self.conn_id,
+            database=self.database,
+            role=self.role,
+            schema=self.schema,
+            warehouse=self.warehouse,
+        )
+
+    @property
+    def conn_type(self):
+        return BaseHook.get_connection(self.conn_id).conn_type
+
+    def execute(self, context: Dict):
+        # handle templating
+        self.handle_params(context)
+
+        # update table name in context based on database
+        context = self._add_templates_to_context(context)
+
+        # render templating in sql query
+        if context:
+            self.sql = self.render_template(self.sql, context)
+
+        # update parameters for snowflake table with fully qualified names
+        self._process_params()
+
+        cursor = self._run_sql(self.sql, self.parameters)
+        if self.handler is not None:
+            return self.handler(cursor)
+        return cursor
```

Comment:
@utkarsharma2 this seems to be missing a fair bit of logic (e.g. handling schemas, handling tables vs. temptables, etc.). Am I missing this somewhere?